src/features/authentication/interceptors/forward-bearer-token.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import { Inject, Injectable } from '@angular/core';
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class ForwardBearerTokenInterceptor implements HttpInterceptor {
+  constructor(private matchUrl: RegExp, @Inject('getToken') private getToken: () => string | null) {}
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    const token: string | null = this.getToken();
+
+    if (shouldForwardBearerToken(this.matchUrl)(token, request.url)) {
+      request = request.clone({
+        setHeaders: {
+          Authorization: `Bearer ${token}`
+        }
+      });
+    }
+
+    return next.handle(request);
```

Comment:
Tu penses que c'est souhaitable d'adopter un format qui ressemble plus à ça ?

```suggestion
    return next.handle(shouldForwardBearerToken(this.matchUrl)(token, request.url) ?
      request.clone({ setHeaders: { Authorization: `Bearer ${token}` } }) :
      request);
```

Reply:
En plus ont peut imaginer de déplacer `request.clone({ setHeaders: { Authorization: `Bearer ${token}` } })` dans une fonction pour augementer la lisibilité, par exemple `requestWithBearerToken(request, token)`

Reply:
👍 
Je sépare shouldForwardBearerToken en 2 pour bénéficier du typeguard sur le token